WordPress/gutenberg
# Add Reverse on Mobile Attribute for Columns Blocks

## What?
Fixes #40706

Adds an attribute `Reverse on mobile` to the Columns Blocks. When selected, the block's column order will be reversed if the screen is narrower than the block's mobile breakpoint.

For Columns Blocks, this means the column order will simply be reversed. 

## Why?

This PR is being submitted along with #56279 because Columns Blocks and Media & Text Blocks are frequently used to display media and content in a zig-zag pattern. For example:
```
[image]   [text ]
[text ]   [image]
[image]   [text ]
```

For this type of pattern, users typically want the content to stack on mobile like this:

```
[image]
[text ]
[image]
[text ]
[image]
[text ]
```

With the current Columns Block behavior, the content ends up looking like this:

```
[image]
[text ]
[text ]
[image]
[image]
[text ]
```

This PR adds the option to reverse the column order on mobile if desired, giving users more flexibility over how responsive content translates to the site's mobile view.

## How?

The Reverse on mobile attribute allows users greater flexibility in determining how their content is translated from unstacked to stacked. Users can choose the order in which the stacked content will appear.

## Testing Instructions

1. Open a post or page.
2. Add at least 3 or 4 Columns Blocks, with 2 columns each.
3. Add media and content in a zig-zag pattern, as described above.
4. Apply the "Reverse on mobile" attribute to every other Columns Block.
5. Observe the stacking behavior in the editor both with and without the attribute toggled.
6. Repeat the previous steps with the Media & Text blocks, except apply the "Reverse on mobile" attribute to all blocks.

### Testing Instructions for Keyboard

1. Insert a Columns Block using existing methods.
2. Select the block using existing methods.
3. Open the Settings sidebar using existing methods.
4. Use the Tab key to highlight Reverse on mobile, and use Space to toggle the attribute.

## Screenshots or screencast 
<img width="1423" alt="Screenshot 2023-11-17 at 12 48 52 AM" src="https://github.com/WordPress/gutenberg/assets/12906034/febc3335-56e0-4593-b7c1-5c56b519ecab">

Screenshot showing the location of the 'Reverse on mobile' attribute in the editor, along with Columns blocks displaying the aforementioned 'zig-zag' pattern.

<img width="1071" alt="Screenshot 2023-11-17 at 12 50 53 AM" src="https://github.com/WordPress/gutenberg/assets/12906034/d098b219-4d47-4931-a30e-f6af614b7e04">

Screenshot showing the appearance of stacked Columns blocks in the editor. The 'Reverse on mobile' attribute has been applied to every other Columns Block so that the 'zig-zag' pattern translates to an appropriate stacked pattern.

<img width="700" alt="Screenshot 2023-10-31 at 10 11 15 PM" src="https://github.com/WordPress/gutenberg/assets/12906034/3f48d6d8-bbbf-471c-8615-4437b3dd6fc0">

Screenshot showing the front end of a page with stacked Columns Blocks, with every other Columns Block having the 'Reverse on mobile' attribute.
